packages/block/src/block.ts

```diff
@@ -234,30 +244,38 @@ export class Block {
   }
 
   /**
-   * Consistency checks and header validation for uncles included
+   * Consistency checks and header validation for uncles included,
    * in the block, if any.
    *
    * Throws if invalid.
    *
+   * The rules of uncles are the following:
+   * Uncle Header is a valid header.
+   * Uncle Header is an orphan, i.e. it is not one of the headers of the canonical chain.
+   * Uncle Header has a parentHash which points to the canonical chain. This parentHash is within the last 7 blocks.
+   * Uncle Header is not already included as uncle in another block.
+   * Header has at most 2 uncles.
+   * Header does not count an uncle twice.
+   *
    * @param blockchain - additionally validate against an @ethereumjs/blockchain instance
    */
   async validateUncles(blockchain: Blockchain): Promise<void> {
     if (this.isGenesis()) {
       return
     }
 
+    // Header has at most 2 uncles
     if (this.uncleHeaders.length > 2) {
       throw new Error('too many uncle headers')
     }
 
+    // Header does not count an uncle twice.
     const uncleHashes = this.uncleHeaders.map((header) => header.hash().toString('hex'))
     if (!(new Set(uncleHashes).size === uncleHashes.length)) {
       throw new Error('duplicate uncles')
     }
 
-    for (const uh of this.uncleHeaders) {
-      await this._validateUncleHeader(uh, blockchain)
-    }
+    await this._validateUncleHeaders(this.uncleHeaders, blockchain)
```

Comment:
On a second look I don't think it makes any sense to keep this double public/private method structure with `validateUncles()` and `_validateUncleHeaders()`, these double `this.uncleHeaders.length` checks (one on > 2, the other on === 0) rather makes this in the current split form more inconsistent, do we want to throw this into one function? 🤔  This double documentation is also a bit "redundancy for nothing" for my taste.